frontend/src/layout/talkList.html

```diff
@@ -0,0 +1,132 @@
+                    <tbody>
+                    <tr ng-repeat="talk in talks | orderBy: 'rating' : true">
+                        <td>
+                            {{ talk.title }}
+                        </td>
+                        <td>
+                            {{ talk.description }}
+                        </td>
+                        <td class="center">
+                            <p>{{ talk.postedBy.username }}</p>
+                        </td>
+                        <th class="center">
+                            {{ talk.rating }}
+                        </th>
+                        <th class="center">
+                            <button type="button" ng-click="rateTalk(talk._id, 1)"
+                                    class="btn btn-default btn-sm btn-success" ng-disabled="talk.rated">Good
+                            </button>
+                            <button type="button" ng-click="rateTalk(talk._id, -1)"
+                                    class="btn btn-default btn-sm btn-danger" ng-disabled="talk.rated">Bad
+                            </button>
+                        </th>
+                    </tr>
+                    </tbody>
+                </table>
+                <h2 ng-if="!isAuthenticated">Please register and login.</h2>
+                <h2 ng-if="isAuthenticated && talks.length == 0">No talk exists. Please create a new one.</h2>
+            </div>
+        </div>
+    </div>
+
+    <script type="text/ng-template" id="loginModal">
+        <div class="modal-header">
+            <h3 class="modal-title">Login</h3>
+        </div>
+        <div class="modal-body">
+            <div class="form-group">
+                <input type="text" ng-model="user.username" class="form-control" placeholder="username">
+            </div>
+
+            <div class="form-group">
+                <input type="password" ng-model="user.password" class="form-control" placeholder="password"
+                       enter-down="login()">
+            </div>
+        </div>
+        <div class="modal-footer">
+            <button ng-click="login()" class="btn btn-default" ng-disabled="!user.username && !user.password">Submit
+            </button>
+        </div>
+    </script>
+
+    <script type="text/ng-template" id="registerModal">
+        <div class="modal-header">
+            <h3 class="modal-title">Register</h3>
+        </div>
+        <div class="modal-body">
+            <div class="form-group">
+                <input type="text" ng-model="user.username" class="form-control" placeholder="username">
+            </div>
+
+            <div class="form-group">
+                <input type="password" ng-model="user.password" class="form-control" placeholder="password"
+                       enter-down="register()">
+            </div>
+        </div>
+        <div class="modal-footer">
+            <button ng-click="register()" class="btn btn-default" ng-disabled="!user.username && !user.password">
```

Comment:
My mistakes for frontend! It should be
```html
ng-disabled="!user.username || !user.password"
```
And same for login and talk posting. Fortunately, the backend handled the null value.